### PR TITLE
adds `values()` method, refactors `showMoreFacets()`

### LINKED ISSUE #11: Add support for values and tuples
Values is very useful for determining min/max, and total count of values (could also be shown in facets). Tuples is necessary for showing co-occurrences (n-way)..

### LINKED ISSUE #56: Make showMoreFacets return a promise
It is now returning `this` to allow chaining, but it makes more sense to return a promise like `search`. You can then do something like:

```javascript
showMore: function showMore(facet, facetName, loading) {
    loading.loading = true;
    return mlSearch.showMoreFacets(facet, facetName, 50).then(function() {
        loading.loading = false;
    });
},
```

### PR DESCRIPTION
**Pull request by joemfb**
closes #11 
re #45, re #50 (need a test case for the bug)
closes #56

### CODE REVIEW

File: src/ml-search.service.js [hunk 1]
```diff
@@ -1128,6 +1121,37 @@
...
+    values: function values(name, params, options) {
+      // TODO: what other conditions
+      if ( !options && params && params.options) {
+        options = params;
+        params = null;
+      }
+
+      params = params || {};
+      params.start = params.start || 1;
+      params.limit = params.limit || 20;
```
Comment (joemfb) [inline src/ml-search.service.js]: by default, `values()` calls are limited to 20 values. Overridable through the `params` argument.

Comment (grtjn) [inline src/ml-search.service.js]: excellent!

File: src/ml-search.service.js [hunk 2]
```diff
@@ -1128,6 +1121,37 @@
...
+
+      if ( !options ) {
+        params.options = this.options.queryOptions;
+      }
+
+      return this.getCombinedQuery(false)
+      .then(function(combined) {
+        combined.options = options;
```
Comment (grtjn) [inline src/ml-search.service.js]: How about getting stored options, checking whether the values are defined, and if not look for facet constraint that could be used to dynamically generate values config (like `showMoreFacets` is doing)?

**Comment by joemfb** [comment]
This doesn't implement your suggestion of first checking for an existing values definition. We can wait till I get a chance to try that, or you could make an additional PR at a later date. Might take me awhile I'm pretty swamped right now ...

Also, I'm not too familiar with the `tuples` options for `/v1/values`. Any reason we'd need another function for that?

Any other outstanding issues?

**Comment by grtjn** [comment]
Not sure we should have a different mlSearch function for tuples. The response will look a little different, but not that much. /v1/values covers both, so perhaps this is good enough for now?

**Comment by grtjn** [comment]
Could be fun if you could dynamically create a tuples definition based on two facet references though.. :)

**Comment by grtjn** [comment]
If you merge this, I could add my other suggestions in a new PR?

**Comment by joemfb** [comment]
Sounds good to me. We can also re-open #11 if changes are needed for tuples support.

**Pull request merged**